src/net/ssl-context.cpp

```diff
@@ -39,14 +41,21 @@ void SslContext::init(std::string ca_bundle_path) {
         throw std::exception();
     }
 
-    if (SSL_CTX_load_verify_locations(ctx, ca_bundle_path.c_str(), NULL) == 0) {
+    if (SSL_CTX_load_verify_locations(ctx, ca_bundle_path.c_str(), NULL) != 1) {
         debug("ssl: failed to load verify location");
         throw std::exception();
     };
 
     SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER, NULL);
 }
 
+Var<SslContext> SslContext::global() {
+    if (global_ == nullptr){
+        global_ = new SslContext();
+    }
+    return Var<SslContext>(global_);
```

Comment:
Hum, yes I do see your point here. We would not be gaining so much in terms of times a new instance is created. I will refactor this to the other singleton pattern you suggest.